cogs/timer.py

```diff
@@ -0,0 +1,188 @@
+from asyncio import sleep
+import random
+
+import discord
+from discord.ext import commands
+from dislash import *
+
+from cogs.help import help
+
+class Timer(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.countdown = 60     # Sekunden bis zum Refresh der Zeitanzeige
+        self.default_names = ["Rapunzel", "Aschenputtel", "Schneewittchen", "Frau Holle", "Schneeweißchen und Rosenrot"]
+        SlashClient(bot)        # Stellt den Zugriff auf die Buttons bereit
+
+    @help(
+        syntax="!timer <learning-time?> <break-time?> <name?>",
+        brief="Deine persönliche Eieruhr",
+        parameters={
+            "learning-time": "Länge der Lernphase in Minuten. Default: 25",
+            "break-time": "Länge der Pausenphase in Minuten. Default: 5",
+            "name": "So soll der Timer heißen. Wird ihm kein Name gegeben, nimmt er sich selbst einen."
+        }
+    )
+    @commands.command(name="timer")
+    async def cmd_timer(self, ctx, lt=25, bt=5, name=None):
+        learning_time = lt
+        break_time = bt
+        name = name if name else random.choice(self.default_names)
+        zeiten = f"{learning_time} Minuten lernen\n{break_time} Minuten Pause"
+        status = ["lernen", lt]
+        angemeldet = [ctx.author]
+
+        button_row = ActionRow(
+            Button(
+                style=ButtonStyle.grey,
+                emoji="🛑",
+                custom_id="beenden"
+            ),
+            Button(
+                style=ButtonStyle.grey,
+                emoji="🔄",
+                custom_id="neustart"
+            ),
+            Button(
+                style=ButtonStyle.grey,
+                emoji="⏩",
+                custom_id="skip"
+            ),
+            Button(
+                style=ButtonStyle.grey,
+                emoji="👍",
+                custom_id="anmelden"
+            ),
+            Button(
+                style=ButtonStyle.grey,
+                emoji="👎",
+                custom_id="abmelden"
+            )
+        )
+
+        async def make_sound(filename):
+            async def disconnect():
+                for vc in self.bot.voice_clients:
+                    await vc.disconnect()
+
+            for user in angemeldet:
```

Comment:
Joint Boty hier, wenn 5 User im gleichen Channel sind 5 mal den Channel und schlägt alarm?

Reply:
nein, beim ersten User der in nem Voice-Channel ist joint er, danach returned die Funktion. das ist falls nicht alle angemeldeten User im Voice sind. Er geht also die Liste durch und sobald er nen User im Voice trifft, joined er und beendet.